src/components/InputV2/Text.tsx

```diff
@@ -0,0 +1,101 @@
+import PropTypes from 'prop-types'
+import { Label } from './Label'
+
+export const TextInput = ({
+  vertical,
+  centered,
+  label,
+  info,
+  error,
+  type,
+  autofocus,
+  disabled,
+  placeholder,
+  controlled,
+  value,
+  unit,
+  onChange,
+  onBlur
+}: TextInputProps) => (
+  <Label disabled={disabled} vertical={vertical} centered={centered}>
+    {vertical ? (
+      <>
+        {label && <span>{label}</span>}
+        {info && <span className='info'>{info}</span>}
+      </>
+    ) : (
+      <div>
+        {label && <span>{label}</span>}
+        {info && <span className='info'>{info}</span>}
+        {error && <span className='error'>{error}</span>}
+      </div>
+    )}
+    <div className='text-input-wrapper'>
+      <input
+        className={error ? 'error' : ''}
+        type={type}
+        autoFocus={autofocus}
+        disabled={disabled}
+        placeholder={placeholder}
+        value={controlled ? value : undefined}
+        defaultValue={controlled ? undefined : value}
+        onChange={e => onChange(e.target.value)}
+        onBlur={e => onBlur(e.target.value)}
+        style={{ paddingRight: `${1 + 0.75 * unit.length}em` }}
+      />
+      {unit && <span className='unit'>{unit}</span>}
+    </div>
+    {vertical && error && <span className='error'>{error}</span>}
+  </Label>
+)
+
+export interface TextInputProps {
+  vertical: boolean
+  centered: boolean
+  label: string
+  info: string
+  error: string
+  type: string
+  autofocus: boolean
+  disabled: boolean
+  placeholder: string
+  controlled: boolean
+  value: string
+  unit: string
+  onChange: (value: string) => void
+  onBlur: (value: string) => void
+}
```

Comment:
For the sake of typescript and consuming apps, should we mark all of these as optional? Are any _actually_ required?

Reply:
They're all passed as default props, so all of them will actually contain valid values.

I think TypeScript / VSCode are smart enough to understand this and won't complain if the apps don't pass them.

I some cases I do actually prefer having a default, so it is always safe to call `onChange()` instead of going `onChange && onChange()` for example

Reply:
Agree with the decision to pass props for the functions, but are you sure typescript doesn't complain given certain configurations? I understand the linter might be smart enough